### DISCUSSION TITLE
Duplicate node names

### DISCUSSION THREAD

**Discussion by swift2plunder** [kdl-org/kdl#164] [Q&A] [answered]
Lua has [this idiom](http://www.lua.org/pil/12.html):
<pre>
    Entry{"Donald E. Knuth",
          "Literate Programming",
          "CSLI",
          1992}
    
    Entry{"Jon Bentley",
          "More Programming Pearls",
          "Addison-Wesley",
          1990}
</pre>

So:
<pre>
Entry "Donald E. Knuth" "Literate Programming" "CSLI" 1992
Entry "Jon Bentley" "More Programming Pearls" "Addison-Wesley" 1990
</pre>

Is that correct KDL?

**Comment by zkat** [comment, marked as answer]
Absolutely! In fact, that's very idiomatic KDL!

I would probably actually change some of those arguments to be properties instead, but at that point, it's a matter of taste (and practicality).

**Reply by swift2plunder** [reply]
In the Lua idiom, those are positional arguments in a function call. Properties *would* be more robust, but that's an example copied from a textbook 🤣 

This answer changed my parser design. For the better, I believe

What would you like the issue to add a test case named (assuming that I didn't simply miss one while searching)?

**Reply by zkat** [reply]
no clue, just figure out a name :)
